### ISSUE TITLE
Language Selection Dropdown not Working

### ISSUE THREAD

**Issue by myusernamedm** [LazZiya/ExpressLocalization#21]
Hi. This was working fine until today, and I have walked through the setup again, compared all files against the downloaded repo, but I can't get it working.

I restructured my Razor pages to add additional @RenderSection 's and the Localize dropdown does nothing when clicked anymore.

Any thoughts are appreciated...its a great package, but I have no idea what had caused the dropdown to stop working. I was only refactoring header code out of the Index page and into _Layout.cshtml.

`<language-nav language-label="NativeName" cookie-handler-url="@Url.Page("/Index", "SetCultureCookie", new { area="", cltr="{0}", returnUrl="{1}" })"></language-nav>`

**Index.cshtml**
```
@page
@model MyApp.Pages.IndexModel
@using LazZiya.ExpressLocalization
@{
    ViewData["Title"] = "MyApp";
}
<body>
    @section header_image{
        <div class="bg-img">
    }
    @section header_content{
        <div class="container-title">
            <div class="block-title block-title1">
                <language-nav language-label="NativeName" cookie-handler-url="@Url.Page("/Index", "SetCultureCookie", new { area="", cltr="{0}", returnUrl="{1}" })"></language-nav>
                <br>
                <a button type="button" class="btn btn-outline-success button-title" asp-page="Register">@_loc["Make A Difference"]</a>
                <h1 class="block-title-text" localize-content>@Model.TitleText</h1>
                <h2 class="block-subtitle-text" localize-content>@Model.SubTitleText1</h2>
                <h2 class="block-subtitle-text" localize-content>@Model.SubTitleText2</h2>
            </div>
            <div class="block-title block-title2 d-none d-md-block d-lg-block d-xl-block"><img src="/image/title_image.png" class="img-fluid"></div>
        </div>
    }
</div>
<main>
    <div class="row_primary">
    </div>
</main>
</body>
```

**_ViewImports.cshtml**
```
@using Microsoft.AspNetCore.Identity
@using MyApp
@using MyApp.Data
@using LazZiya.ExpressLocalization
@inject ISharedCultureLocalizer _loc
@namespace MyApp.Pages
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
@using LazZiya.TagHelpers
@addTagHelper *, LazZiya.TagHelpers
@addTagHelper *, LazZiya.ExpressLocalization
```

**Startup.cs**
Has everything as per the instructions.

**Index.cshtml.cs**
```
        public IActionResult OnGetSetCultureCookie(string cltr, string returnUrl)
        {
            Response.Cookies.Append(
                CookieRequestCultureProvider.DefaultCookieName,
                CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(cltr)),
                new CookieOptions { Expires = DateTimeOffset.UtcNow.AddYears(1) }
            );

            return LocalRedirect(returnUrl);
        }
```
**_Layout.cshtml**
```
@using LazZiya.ExpressLocalization
@inject ISharedCultureLocalizer _loc
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>@ViewData["Title"] | @_loc["Subtitle"]</title>
    <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
    <link rel="stylesheet" href="~/css/style.css" />
</head>

<body>
    <header>
        @RenderSection("header_image", required: false)
        <nav class="navbar navbar-expand-lg navbar-light">
            <div class="container">
                <a class="navbar-brand mt-20" asp-page="Index">
                    <img src="~/image/circle_logo.png" width="40" height="40" style="margin:0px 20px;" alt="Logo Image">
                </a>
                <button class="navbar-toggler" type="button" data-toggle="collapse"
                        data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span class="navbar-toggler-icon"></span>
                </button>
                <div class="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul class="navbar-nav mr-auto">
                        <li class="nav-item active">
                            <a class="nav-link" style="color: rgba(255, 140, 0, 0.781)" asp-page="Index">@_loc["Home"]</a>
                        </li>
                        <li class="nav-item active">
                            <a class="nav-link" style="color: green" asp-page="Features">@_loc["Features"]</a>
                        </li>
                        <li class="nav-item">
                            <a class="nav-link" style="color: green" asp-page="About">@_loc["About Us"]</a>
                        </li>
                        <li class="nav-item">
                            <a class="nav-link" style="color: green" asp-page="HelpCenter">@_loc["Help Center"]</a>
                        </li>
                        <partial name="_LoginPartial" />
                    </ul>
                </div>
            </div>
        </nav>
        @RenderSection("header_content", required:false)
    </header>
    <main role="main">
        @RenderBody()
    </main>
```

**Comment by myusernamedm** [comment]
**Update:** the dropdown is not working on any of my pages, regardless of where I place                     `<language-nav language-label="NativeName" cookie-handler-url="@Url.Page("/Index", "SetCultureCookie", new { area="", cltr="{0}", returnUrl="{1}" })"></language-nav>`

**Comment by LazZiya** [comment]
Hi @myusernamedm ,

Can you please give me more details to figure out the problem;

- Which version of `ExpressLocalization` you are using?
- Which .net platform the project is targetting?
- What is the project type (MVC, Razor Pages)?
- When you look at the rendered page source code, do you see any generated link for the languages?
````html
http://localhost:1234/xx/Index?cltr=en&returnUrl=/en&handler=SetCultureCookie
````
- Does it work when you put the `<language-nav>` inside `_Layout.cshtml` right above the `_LoginPartial`?
````html
<language-nav ... ></language-nav>
<partial name="_LoginPartial" />
````
- Last but not least, if you look at the cookies, do you see language and consent cookies? 
![image](https://user-images.githubusercontent.com/22244584/82138621-82e02f00-982a-11ea-9ef5-fe7855ec6256.png)
 Notice that `.AspNet.Consent` cookie must be accepted **Yes**, and the `.AspNetCore.Culture` cookie must hold the selected cuture value

**Comment by myusernamedm** [comment]
Hi,
I reverted all my commits and re-implemented my changes one by one to try to find the problem. But it has been working fine. I missed something but just can't figure out what! At least it is working again. Very frustrating not knowing what caused the issue. If I typed the culture into the URL manually, it worked fine, it was just the dropdown component did nothing when clicked. Very strange, but working again now...
